packages/default/scss/button/_layout.scss

```diff
@@ -354,3 +276,69 @@
     }
 
 }
+
+
+
+@include exports( "button/layout/thin-background" ) {
+    $flat-button-hover-opacity: .08 !default;
+    $flat-button-focused-opacity: .12 !default;
+    $flat-button-active-opacity: .16 !default;
+    $flat-button-selected-opacity: .20 !default;
+
+    // Thin background
+    .k-button {
+
+        &::before {
+            content: "";
+            background: currentColor;
+            opacity: 0;
+            display: none;
+            pointer-events: none;
+            position: absolute;
+            left: -$button-border-width;
+            right: -$button-border-width;
+            top: -$button-border-width;
+            bottom: -$button-border-width;
+            z-index: 0;
+            transition: opacity .2s ease-in-out;
+        }
+
+        // Hovered state
+        &:hover,
+        &.k-state-hover {
+            &::before {
+                opacity: $flat-button-hover-opacity;
+            }
+        }
+
+        // Focused state
+        &:focus,
+        &.k-state-focused {
+            &::before {
+                opacity: $flat-button-focused-opacity;
+            }
+        }
+        &.k-no-focus:not(:hover),
+        &.k-no-focus:not(.k-state-hover) {
```

Comment:
This will not work as expected, because `.k-state-hover` will be positive for the first selector and `:hover` for the second... In other words, it will catch all selectors...